crates/fuel-core/src/graphql_api/api_service.rs

```diff
@@ -373,17 +376,43 @@ where
     ))
 }
 
-async fn render_graphql_playground(
+/// Single initialization of the GraphQL playground HTML.
+/// This is because the rendering and replacing is expensive
+static GRAPHQL_PLAYGROUND_HTML: OnceLock<Arc<String>> = OnceLock::new();
+
+fn _render_graphql_playground(
     endpoint: &str,
     subscription_endpoint: &str,
-) -> impl IntoResponse + use<> {
-    Html(
-        GraphiQLSource::build()
+) -> impl IntoResponse + Send + Sync {
+    let html = GRAPHQL_PLAYGROUND_HTML.get_or_init(|| {
+        let raw_html = GraphiQLSource::build()
             .endpoint(endpoint)
             .subscription_endpoint(subscription_endpoint)
             .title("Fuel Graphql Playground")
-            .finish(),
-    )
+            .finish();
+
+        // this may not be necessary in the future,
+        // but we need it to patch: https://github.com/async-graphql/async-graphql/issues/1703
+        let raw_html = raw_html.replace(
+            "https://unpkg.com/graphiql/graphiql.min.js",
+            "https://unpkg.com/graphiql@3/graphiql.min.js",
+        );
+        let raw_html = raw_html.replace(
+            "https://unpkg.com/graphiql/graphiql.min.css",
+            "https://unpkg.com/graphiql@3/graphiql.min.css",
+        );
```

Comment:
while we could get away with a singular replace, i.e 
```suggestion
        let raw_html = raw_html.replace(
            "https://unpkg.com/graphiql/graphiql.min.",
            "https://unpkg.com/graphiql@3/graphiql.min.",
        );
```
i would rather have it be explicit about which assets we are pinning, and since this part of the code only runs once, it seems ok imo

Reply:
Agreed.